api/node/index.ts

```diff
@@ -469,6 +471,60 @@ export function loadFile(filePath: string, options?: LoadFileOptions): Object {
                 }
             });
 
+            // globals
+            instance!.definition().globals.forEach((globalName) => {
+                if (componentHandle[globalName] !== undefined) {
+                    console.warn("Duplicated property name " + globalName);
+                } else {
+                    let globalObject = new SlintGlobal;
+
+                    instance!.definition().globalProperties(globalName).forEach((prop) => {
+                        let propName = prop.name.replace(/-/g, "_");
+
+                        if (globalObject[propName] !== undefined) {
+                            console.warn("Duplicated property name " + propName + " on global " + global);
+                        } else {
+                            Object.defineProperty(globalObject, propName, {
+                                get() {
+                                    return instance!.getGlobalProperty(globalName, prop.name);
+                                },
+                                set(value) {
+                                    instance!.setGlobalProperty(globalName, prop.name, value);
+                                },
+                                enumerable: true,
+                            });
+                        }
+                    });
+
+                    instance!.definition().globalCallbacks(globalName).forEach((cb) => {
+                        let callbackName = cb.replace(/-/g, "_");
+
+                        if (globalObject[callbackName] !== undefined) {
+                            console.warn("Duplicated property name " + cb + " on global " + global);
+                        } else {
+                            Object.defineProperty(globalObject, cb.replace(/-/g, "_"), {
+                                get() {
+                                    return function () {
+                                        return instance!.invokeGlobal(globalName, cb, Array.from(arguments));
+                                    };
```

Comment:
I realise that we have a bug here with our semantics. In the getters for our callbacks we always return a new function object. That's not correct, I think. If you look at this [test case](https://www.typescriptlang.org/play?#code/FAGwpgLgBBYM4QFxQIYDsCeUC8UDeAvsMAGYD2ATlABTjQDGKIIARivQNZRklQDaAIhYgUACwEAaKAPJkBAXQCU+YFCgB5FgCsw9CADoAJmBIBLNGAAKFMgAcwFCBmqwEUxszacpeVWv8A5pDUyr7+4VAUkACuFGhQJNFoeqZkaCH4kTFxUB6s7BwA3FBEESV+aqVQioXARMR0CaYUCDgw8AbCYrWNcLpphm2unSKitfRpcGTg+iBkAdRmLdDYq1B9E2iGisCbUzNzC0sIITtAA) you'll see that it prints false, when it should print true:

```

let test: any = {}

for (let callback of ["blah", "foo"]) {
  Object.defineProperty(test, callback, {
      get() {
        return function() { return callback; }
      }
    }
  );
}


let first = test.blah;
let second = test.blah;
console.log(first === second)
console.log(first())
```

The first is to not always return a new closure, like [this](https://www.typescriptlang.org/play?#code/FAGwpgLgBBYM4QFxQIYDsCeUC8UDeAvsMAGYD2ATlABTjQDGKIIARivQNZRklQDaAIhYgUACwEAaKAPJkBAXQCU+YFCh0o9FgH0AlgFsADiBxQSAVzT0IusmmrK8qtVAqRzFNJqat2HANzORGoA8iwAVmDWAHQAJmAkumhgAAoUZIZgFBAY1LAIUozMbJxSTi4uAOaQDioVFW4QHl5aekYggfVQwS49ioFExBqJFAim+RDRwmKBGnBRdrHj8JPTooH0dnBk4NEgZJXUI2PYp1Dzm2ixisCX27v7h8cQDjdAA):

```

let test: any = {}

for (let callback of ["blah", "foo"]) {
  let cb_impl = function() {
    return callback;
  }
  Object.defineProperty(test, callback, {
      get() {
        return cb_impl;
      }
    }
  );
}


let first = test.blah;
let second = test.blah;
console.log(first === second)
console.log(first())
```

This is not in scope of this PR, but we should really have a test for it and fix it.

Reply:
Mh yes that's a problem. Without the function wrapping it didn't work because the getter must return a function that you can call.

Reply:
But nevertheless we have still the problem with callbacks and gc. So I think that are two problems that should solved ad once. Or what do you think?

Reply:
Good point, you're right.